Assets/Scripts/Input/Gameplay/GameplayMouseAndKeyboardInputControl.cs

```diff
@@ -22,29 +21,33 @@ public class GameplayMouseAndKeyboardInputControl : TOSGMonoBehaviourBase, IGame
 
         #region - - - - - - Fields - - - - - -
 
-        private IPauseActionHandler m_PauseAction;
+        // Player behavior handlers
         private IPlayerAttackHandler m_PlayerAttackHandler;
         private IPlayerMovement m_PlayerMovement;
         private IPlayerTargetTracking m_PlayerTargetTracking;
         private IPlayerSpecialAction m_PlayerSpecialAction;
         private IPlayerViewOrientationHandler m_PlayerViewOrientationHandler;
-        
+
+        // Menu behavior handlers
+        private IPauseActionHandler m_PauseActionHandler;
+
         private bool m_IsInputActive;
 
         #endregion Fields
-
+        
         #region - - - - - - Lifecycle Methods - - - - - -
 
         private void Start()
         {
-            GameState gameState = GameManager.instance.GameState;
-
-            this.m_PauseAction = gameState.PauseMenu.GetComponent<IPauseActionHandler>();
-            this.m_PlayerMovement = gameState.ActivePlayer.GetComponent<IPlayerMovement>();
-            this.m_PlayerTargetTracking = gameState.ActivePlayer.GetComponent<IPlayerTargetTracking>();
-            this.m_PlayerAttackHandler = gameState.ActivePlayer.GetComponent<IPlayerAttackHandler>();
-            this.m_PlayerSpecialAction = gameState.ActivePlayer.GetComponent<IPlayerSpecialAction>();
-            this.m_PlayerViewOrientationHandler = gameState.ActivePlayer.GetComponent<IPlayerViewOrientationHandler>();
+            var _GameState = GameManager.instance.GameState;
+            this.m_PauseActionHandler = _GameState.SessionUser.GetComponent<IPauseActionHandler>();
+            
+            var _ActivePlayer = GameManager.instance.SceneManager.SceneState.ActivePlayer;
+            this.m_PlayerMovement = _ActivePlayer.GetComponent<IPlayerMovement>();
+            this.m_PlayerTargetTracking = _ActivePlayer.GetComponent<IPlayerTargetTracking>();
+            this.m_PlayerAttackHandler = _ActivePlayer.GetComponent<IPlayerAttackHandler>();
+            this.m_PlayerSpecialAction = _ActivePlayer.GetComponent<IPlayerSpecialAction>();
+            this.m_PlayerViewOrientationHandler = _ActivePlayer.GetComponent<IPlayerViewOrientationHandler>();
 
```

Comment:
nit: lots of dependency on the GameManager here, could be worth inverting this to inject the ActivePlayer instead

Reply:
Inverted the control by using a command to handle operations to initializing the dependencies of the GameplayInput.